src/libstore/store-api.cc

```diff
@@ -214,7 +216,11 @@ StorePath Store::addToStore(
     auto source = sinkToSource([&](Sink & sink) {
         dumpPath(path, sink, fsm, filter);
     });
-    return addToStoreFromDump(*source, name, fsm, method, hashAlgo, references, repair);
+    LengthSource lengthSource(*source);
+    auto storePath = addToStoreFromDump(lengthSource, name, fsm, method, hashAlgo, references, repair);
+    if (lengthSource.total >= settings.largePathWarningThreshold)
+        warn("copied large path '%s' to the store (%d bytes)", path, lengthSource.total);
```

Comment:
Minor suggestion:

Do we already have a way to convert the raw number of bytes into a human-readable number?

At least for me, reading "32819029 bytes" doesn't communicate nearly as much information as "32.819029 MB", or "74789190238 bytes" and "74.789190238 GB"

Reply:
Done.